app/src/main/java/de/westnordost/streetcomplete/quests/surface/AddRoadSurfaceForm.kt

```diff
@@ -5,19 +5,15 @@ import de.westnordost.streetcomplete.R
 import de.westnordost.streetcomplete.quests.AImageListQuestAnswerFragment
 import de.westnordost.streetcomplete.view.image_select.Item
 
-class AddRoadSurfaceForm : AImageListQuestAnswerFragment<String, SurfaceAnswer>() {
-    override val items: List<Item<String>>
-        get() =
-            (PAVED_SURFACES + UNPAVED_SURFACES + GROUND_SURFACES).toItems() +
-                Item("paved", R.drawable.surface_paved, R.string.quest_surface_value_paved, null, listOf()) +
-                Item("unpaved", R.drawable.surface_unpaved, R.string.quest_surface_value_unpaved, null, listOf()) +
-                Item("ground", R.drawable.surface_ground, R.string.quest_surface_value_ground, null, listOf())
+class AddRoadSurfaceForm : AImageListQuestAnswerFragment<Surface, SurfaceAnswer>() {
+    override val items: List<Item<Surface>>
+        get() = (PAVED_SURFACES + UNPAVED_SURFACES + GROUND_SURFACES + GENERIC_SURFACES).toItems()
 
     override val itemsPerRow = 3
 
-    override fun onClickOk(selectedItems: List<String>) {
+    override fun onClickOk(selectedItems: List<Surface>) {
         val value = selectedItems.single()
-        if (value == "paved" || value == "unpaved") {
+        if (value.shouldDescribe) {
```

Comment:
Hmm, not sure if this is an improvement. I'd rather have the check here or maybe rename `shouldDescribe` to `isGeneric`?

Reply:
This check is duplicated across all surface quest forms, that's why I extracted it.

First, I just had `GENERIC_SURFACES.contains(value)`, but then I saw [`ac55db8` (#2078)](https://github.com/streetcomplete/StreetComplete/pull/2078/commits/ac55db8e5f1013ec9a5b986a6758fba8baa12202), which explicitly excludes `ground`.

I think `isGeneric` sounds a bit like a shorthand for `GENERIC_SURFACES.contains(value)`.

Reply:
Hm thats true. But maybe then `shouldBeDescribed`?